lib/HLSL/DxilTargetTransformInfo.h

```diff
@@ -0,0 +1,43 @@
+//===-- DxilTargetTransformInfo.h - DXIL specific TTI -------*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+/// \file
+/// This file declares a TargetTransformInfo analysis pass specific to the DXIL.
+/// Only implemented isSourceOfDivergence for DivergenceAnalysis.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "llvm/CodeGen/BasicTTIImpl.h"
+
+namespace hlsl {
+class DxilModule;
+class OP;
+}
+
+namespace llvm {
+
+class DxilTTIImpl final : public BasicTTIImplBase<DxilTTIImpl> {
+  typedef BasicTTIImplBase<DxilTTIImpl> BaseT;
+  typedef TargetTransformInfo TTI;
+  friend BaseT;
+  hlsl::OP *m_pHlslOP;
+  bool m_isThreadGroup;
```

Comment:
This member variable looks a bit odd. Seems like it is more about divergence analysis than the target. Since we always set it to true (in this patch). Lets hold off on this for now and add it when we see the use case. Perhaps we need to extend divergence analysis interface instead.